packages/node_modules/samples/browser-call-with-screenshare/app.js

```diff
@@ -100,101 +69,146 @@ function connect() {
+      document.getElementById('self-view').srcObject = null;
     }
-    else {
-      document.getElementById('self-screen').srcObject = undefined;
-      document.getElementById('screenshare-tracks').innerHTML = 0;
+    if (media.type === 'remoteVideo') {
+      document.getElementById('remote-view-video').srcObject = null;
+    }
+    if (media.type === 'remoteAudio') {
+      document.getElementById('remote-view-audio').srcObject = null;
     }
   });
 
-  call.on('remoteMediaStream:change', () => {
-    // Ok, yea, this is a little weird. There's a Chrome behavior that prevents
-    // audio from playing from a video tag if there is no corresponding video
-    // track.
-    [
-      'audio',
-      'video'
-    ].forEach((kind) => {
-      if (call.remoteMediaStream) {
-        const track = call.remoteMediaStream.getTracks().find((t) => t.kind === kind);
-
-        if (track) {
-          document.getElementById(`remote-view-${kind}`).srcObject = new MediaStream([track]);
-        }
-      }
-    });
+  // Of course, we'd also like to be able to end the call:
+  document.getElementById('hangup').addEventListener('click', () => {
+    meeting.leave();
   });
 
-  // Once the call ends, we'll want to clean up our UI a bit
-  call.on('inactive', () => {
-    // Remove the streams from the UI elements
-    document.getElementById('self-view').srcObject = undefined;
-    document.getElementById('remote-view-audio').srcObject = undefined;
-    document.getElementById('remote-view-video').srcObject = undefined;
+  meeting.on('all', (event) => {
+    console.log(event);
+  });
+}
 
-    // And unset the call object
-    activeCall = undefined;
+// Join the meeting and add media
+function joinMeeting(meeting) {
+  return meeting.join().then(() => {
+    const mediaSettings = {
+      receiveVideo: true,
+      receiveAudio: true,
+      receiveShare: false,
+      sendVideo: true,
+      sendAudio: true,
+      sendShare: false
+    };
+
+    return meeting.getMediaStreams(mediaSettings).then((mediaStreams) => {
+      const [localStream, localShare] = mediaStreams;
+
+      meeting.addMedia({
+        localShare,
+        localStream,
+        mediaSettings
+      });
+    });
   });
 }
 
 // In order to simplify the state management needed to keep track of our button
-// handlers, we'll rely on the current call global object and only hook up event
+// handlers, we'll rely on the current meeting global object and only hook up event
 // handlers once.
 
-document.getElementById('hangup').addEventListener('click', () => {
-  if (activeCall) {
-    activeCall.hangup();
-  }
-});
-
 document.getElementById('share-screen').addEventListener('click', () => {
-  if (activeCall) {
-    activeCall.startScreenShare();
+  if (activeMeeting) {
+    const mediaSettings = {
+      receiveShare: false,
+      sendShare: true
+    };
+
+    activeMeeting.getMediaStreams(mediaSettings)
+      .then(([, localShare]) => activeMeeting.updateShare({
```

Comment:
What does this syntax denote? I've seen it but forget. 🤔

Reply:
It is the same as doing `const localShare = mediaSettingsResultArray[1]`;

Reply:
TIL

Reply:
Is this too much JS magic for a demo?

Reply:
It might be, but if you leave a comment explaining what's happening then it's a teaching moment 😄. I do something similar with the examples in the README
```js
// `[, localShare]` is grabbing index 1 from the `mediaSettingsResultsArray` and storing it in a variable called localShare.
```
^something like that